File: frontend-vue/domain/proposal.ts
import { Movie } from "./movie";

export class Proposal {
  id: string;
  movie: Movie;
  votes: number;

  constructor(
    id: string,
    movie: Movie,
    votes: number,
  ) {
    this.id = id;
    this.movie = movie;
    this.votes = votes;
  }
}

File: backend/models/proposal.go
package models

type Proposal struct {
	Id        int64    `json:"id" pg:"id,pk" validate:"required"`
	MovieId   int      `json:"movieId" pg:",unique:session_movie" validate:"required"`
	Comment   string   `json:"comment"`
	SessionId int64    `json:"sessionId" pg:",unique:session_movie" validate:"required"`
	Session   *Session `json:"session" pg:"rel:has-one"`
	Votes     []*Vote  `json:"votes" pg:"rel:has-many"`
	Resource
} //@name Proposal

type CreateProposalRequest struct {
	SessionId int64  `json:"sessionId" validate:"required"`
	MovieId   int    `json:"movieId" validate:"required"`
	Comment   string `json:"comment"`
} //@name CreateProposalRequest
